### PR TITLE
AS-106: Implement fetching token URL dynamically from OIDC Configuration

### PR DESCRIPTION
**Pull request by rahulv-dev**

### CODE REVIEW

File: src/runtime.ts [hunk 1]
```diff
@@ -26,6 +27,7 @@ export class ApiClient {
     private middleware: Middleware[];
     public sdkVersion: string = null;
     private accessTokenMap: Map<string, TokenMetadata> = new Map();
+    public tokenUrl = null;
```
Comment (jwenger100) [inline src/runtime.ts:30]: could you put type here? public tokenUrl: string

File: src/runtime.ts [hunk 2]
```diff
@@ -133,6 +136,22 @@ export class ApiClient {
         return response;
      }
 
+    private async populateTokenUrl(openIdConnectUrl: string) {
+        const { environment, testTokenUrl } = this.configuration;
+        if(environment === AvaTaxEnvironment.Test ) {
+            this.tokenUrl = testTokenUrl;
+        } else if (this.tokenUrl == null || this.tokenUrl.length == 0) {
+            const tokenUrlResponse = await this.getTokenUrl(openIdConnectUrl);
```
Comment (jwenger100) [inline src/runtime.ts:144]: did we decide on fallback logic if this call fails? or we could put error handling here to tell the caller what failed for now

File: src/runtime.ts [hunk 3]
```diff
@@ -133,6 +136,22 @@ export class ApiClient {
         return response;
      }
 
+    private async populateTokenUrl(openIdConnectUrl: string) {
+        const { environment, testTokenUrl } = this.configuration;
+        if(environment === AvaTaxEnvironment.Test ) {
```
Comment (jwenger100) [inline src/runtime.ts:141]: nit: space between if and (, no space between test enum and paren

File: src/runtime.ts (hunk 1 above)
Comment (rahulv-dev) [reply src/runtime.ts]: Done

File: src/runtime.ts (hunk 2 above)
Comment (rahulv-dev) [reply src/runtime.ts]: No we didn't. I have added error handling and kept it similar to Java and C Sharp SDK.

File: src/runtime.ts (hunk 3 above)
Comment (rahulv-dev) [reply src/runtime.ts]: Done

**Pull request merged**